src/client.cpp

```diff
@@ -64,33 +67,268 @@ void set_client_timer(struct itimerval *timer) {
+void storeHistogram(int binSize, std::map<int, int> &activeBins, int minValue, int maxValue) {
+    const int leftOutlierBinIndex = getLeftOutlierBinIndexReserved();
+    const int rightOutlierBinIndex = getRightOutlierBinIndexReserved();
+    const int lowerRange = s_user_params.histogram_lower_range;
+    const int upperRange = s_user_params.histogram_upper_range;
+    int startBinEdge = 0;
+    int frequency = 0;
+    // Align columns with the following variables
+    const int whitespaceAfterMaxBinMessage = 20;
+    int upperRangeDigits = getNumberOfDigits(upperRange);
+    int maxValueDigits = getNumberOfDigits(maxValue);
+    int minValueDigits = getNumberOfDigits(minValue);
+    int columnWidth = maxValueDigits + 1 + upperRangeDigits + whitespaceAfterMaxBinMessage;
+    int columnWidthLeftOutlier = columnWidth - 1 - minValueDigits; // - 1 for "-"
+    int columnWidthRightOutlier = columnWidth - 1 - upperRangeDigits; // - 1 for "-"
+
+    std::map<int, int>::iterator itr;
+    FILE *f = g_pApp->m_const_params.fileFullLog;
+
+    fprintf(f, "histogram was built using the following parameters: "
+            "--h_bin_size_us=%d --h_lower_range_us=%d --h_upper_range_us=%d\n",
+            (int)g_pApp->m_const_params.histogram_bin_size,
+            (int)g_pApp->m_const_params.histogram_lower_range,
+            (int)g_pApp->m_const_params.histogram_upper_range);
+    fprintf(f, "------------------------------\n");
+    fprintf(f, "bin (usec)            frequency\n");
+    for(itr = activeBins.begin(); itr != activeBins.end(); ++itr) {
+        frequency = itr->second;
+        startBinEdge = (itr->first - 1) * binSize + lowerRange;
+        if (itr->first == leftOutlierBinIndex) {
+            fprintf(f, "%d-%-*d%d\n", minValue, columnWidthLeftOutlier, lowerRange, frequency);
+        } else if (itr->first == rightOutlierBinIndex) {
+            fprintf(f, "%d-%-*d%d\n", getStartOfRightOutlierBin(), columnWidthRightOutlier,
+                    maxValue, frequency);
+        } else {
+            fprintf(f, "%-*d%d\n", columnWidth, startBinEdge, frequency);
+        }
+    }
+    fprintf(f, "------------------------------\n");
+}
+
+//------------------------------------------------------------------------------
+/*  Display histogram to fit on terminal screen width (frequency rounded up) */
+void printHistogram(int binSize, std::map<int, int> &activeBins, int minValue, int maxValue) {
+    const int leftOutlierBinIndex = getLeftOutlierBinIndexReserved();
+    const int rightOutlierBinIndex = getRightOutlierBinIndexReserved();
+    const int lowerRange = s_user_params.histogram_lower_range;
+    const int upperRange = s_user_params.histogram_upper_range;
+    const std::string prefixToHistogramDisplay ("sockperf: ");
+    int maxFrequency = 0;
+    int currFrequency = 0;
+    int terminalWidth = 0;
+    int scalingUnit = 0;
+    int maxDisplayWidth = 0;
+    int maxStartBinDigits = getNumberOfDigits(maxValue - binSize);
```

Comment:
`maxValue` may be less or equal to `binSize`. In these cases `log10` returns NaN or -inf and number of digits is incorrect (< 0). This leads to "freeze" in column formatting code.
```suggestion
    int maxStartBinDigits = getNumberOfDigits(getStartOfRightOutlierBin());
```